### PR TITLE
add generators for ts and npm publish

### PR DESCRIPTION
**Pull request by jacobheric**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -51,6 +51,24 @@ generate/go:
 
 	@echo Successfully compiled protos
 
+.PHONY: generate/ts
+generate/ts: description = Compile protobuf schema descriptor and generate types for Typescript
+generate/ts: clean/ts
+generate/ts:
+	docker run --platform linux/amd64 --rm -v ${PWD}:${PWD} -w ${PWD} ${PROTOC_IMAGE} \
```
Comment (dselans) [inline Makefile:58]: Do you need to `mkdir -p build/ts`?

**Review by dselans — APPROVED** [review]
One minor comment.

File: Makefile (hunk 1 above)
Comment (dselans) [reply Makefile:58]: Actually, I think `/build/ts/descriptor-sets`?

Comment (jacobheric) [reply Makefile:60]: Yup, fixed!

**Pull request merged**